index.js

```diff
@@ -138,6 +140,36 @@ function findAllDependencies(file, knownDependencies) {
   });
 }
 
+// write compiled Elm to a string output
+// callback takes (err, data), where data is a Buffer of
+// the compiled text.
+// If you want html instead of js, use options object to set
+// output to a html file instead
+// creates a temp file and deletes it after reading
+function compileToString(sources, options, callback){
+  if (typeof options.output === "undefined"){
+    options.output = '.js';
+  }
+
+  temp.open({ suffix: options.output }, function(err, info){
+    if (err){
+      return callback(err, null);
+    }
+
+    options.output = info.path;
+
+    compile(sources, options)
+      .on("close", function(exitCode){
+        console.log(exitCode, info.path);
+        fs.readFile(info.path, function(err, data){
+          callback(err, data);
+          temp.cleanupSync();
```

Comment:
Any reason not to do this before the callback? Seems like we're pretty conclusively done with the file before we fire the callback.